src/Helpers/JsonLd.php

```diff
@@ -18,4 +29,139 @@ public static function getType($thing)
+            // and camel-cased
+            $attrName = Str::camel(substr($methodName, 3));
+
+            // Attribute value is the result of calling $methodName on $obj
+            $attrValue = $obj->$methodName();
+
+            if(is_array($attrValue)) {
+                // If attribute value is an array,
+                // get data for serialization from each of the item.
+                foreach($attrValue as $idx => $item) {
+                    if(is_object($item)) {
+                        // Get fully qualified namespace of the item's class name
+                        $fq_classname = "\\".get_class($item);
+
+                        $item = self::prepareDataForSerialization(
+                            $item
+                        );
+                    }
+
+                    $attrValue[$idx] = $item;
+                }
+            } else if(is_object($attrValue)) {
+                // If attribute value is an object,
+                // get the data for the individual object
+
+                // Get fully qualified namespace of the item's class name
+                $fq_classname = "\\".get_class($attrValue);
+
+                switch ($fq_classname) {
+                    case "\\DateInterval":
+                        // Get interval spec string, e.g. "P1D"
+                        $attrValue = DateIntervalHelper::specString($attrValue);
+                        break;
+                    case "\\DateTime":
+                        // Get ISO 8601 date time representation,
+                        // e.g. "2019-01-01T00:00:00-08:00"
+                        $attrValue = DateTimeHelper::iso8601($attrValue);
+                        break;
+                    default:
+                        $attrValue = self::prepareDataForSerialization(
+                            $attrValue
+                        );
+                }
+            }
+
+            $data[$attrName] = $attrValue;
+        }
+
+        // Remove empty elements
+        return array_filter(
+            $data,
+            function($value) {
+                if(is_array($value) === TRUE && count($value) === 0) {
+                    // Filter out empty arrays
+                    return false;
+                }
+
+                // Filter out null values and empty strings
+                return $value !== null && $value !== "";
+            }
+        );
+    }
+
+    /**
+     * Remove all "@context" attributes from the given JSON-LD string,
+     * leaving only the first occurrence.
+     *
+     * @param string $json
+     * @return string
+     * @see https://github.com/openactive/OpenActive.NET/blob/master/OpenActive.NET/OpenActiveSerializer.cs#L113 For .NET implementation
+     */
+    public static function removeAllButFirstContext($json)
+    {
+        // OpenActive beta's context JSON-LD representation
+        $openActiveContextPropertyWithBeta = "\"@context\":[\"https:\\/\\/openactive.io\\/\",\"https:\\/\\/openactive.io\\/ns-beta\"],";
```

Comment:
I'd personally use an array here and json_encode it and strip off the `{`, `}` after... should be a little more readable

Reply:
Agree - thought about it but didn't implement it in the end, will implement now :)

Reply:
PR ready for you my friend https://github.com/openactive/models-php/pull/20